core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -401,9 +411,6 @@ class GroupCoordinator(val brokerId: Int,
             val memberMetadata = group.get(memberId)
             responseCallback(memberMetadata.assignment, Errors.NONE)
             completeAndScheduleNextHeartbeatExpiration(group, group.get(memberId))
-
-          case Dead =>
```

Comment:
Could we just leave this unchanged? The non-exhaustive check means we'll have warnings again.

Reply:
Fine. I'm going to revert this as well as line 536 below.